R/loopy.R

```diff
@@ -70,36 +76,105 @@ loopy <- function(resamples, grid, static) {
+          sub_grid <- current_prd[, sub_nm]
+
+          # The assigned submodel parameter (from min_grid()) is in the
+          # current grid. Remove that and add the one that we are predicting on
+
+          current_grid <- current_grid |>
+            dplyr::select(-dplyr::all_of(sub_nm)) |>
+            rebind_grid(current_prd)
+
+          # Remove the submodel column since it is in the currrent grid.
+          current_pred <- predict_all_types(current_wflow, static, sub_grid) |>
+            dplyr::select(-dplyr::all_of(sub_nm))
+
+        } else {
+          current_pred <- predict_all_types(current_wflow, static)
         }
-        pred <- remove_log_notes(pred)
-      }
-      # ------------------------------------------------------------------------
-      # Allocate predictions to an overall object
 
-      pred_iter <- pred_iter + 1
-      pred_reserve <- update_reserve(pred_reserve, pred_iter, pred, nrow(grid))
+        has_post <- has_tailor(current_wflow)
+        num_pst_iter <- nrow(current_prd$post_stage[[1]])
+
+        # ----------------------------------------------------------------------
+        # Iterate over postprocessors
+
+        current_predict_grid <- current_grid
+
+        for (pst in seq_len(num_pst_iter)) {
+          # cli::cli_inform("-- Postprocessing {pst} of {num_pst_iter}")
+
+          if (has_post) {
+            current_pst <- current_prd$post_stage[[1]][pst, ]
+            post_grid <- current_pst
+
+            current_post_grid <- rebind_grid(current_predict_grid, current_pst)
+
+            # make data for prediction (TODO maybe make a function)
+            if (has_tailor_estimated(current_wflow)) {
+              tailor_train_data <- predict_all_types(
+                current_wflow,
+                static,
+                predictee = "calibration"
+              )
+            } else {
+              tailor_train_data <- current_pred[0,]
+            }
+
+            post_fit <- train_post(
+              current_wflow,
+              predictions = tailor_train_data,
+              grid = post_grid
+            )
+
+            post_pred <- predict(post_fit, current_pred)
+
+            current_wflow <- set_workflow_tailor(current_wflow, post_fit)
+            final_pred <- dplyr::bind_cols(post_pred, current_post_grid)
+
+          } else {
+            # No postprocessor so just use what we have
+            final_pred <- dplyr::bind_cols(current_pred, current_predict_grid)
+          }
+
+          # --------------------------------------------------------------------
+          # Allocate predictions to an overall object
+
+          pred_iter <- pred_iter + 1
+          # cli::cli_inform("-- -- Allocation {pred_iter} of {nrow(grid)}")
+          # TODO We might not be able to predict ahead of time how many rows should
+          # be in the reserve
+          # pred_reserve <- update_reserve(pred_reserve, pred_iter, final_pred, nrow(grid))
```

Comment:
I punted on using the reserve for now. The initial cache size depends on the inner split (if any) so the logic needs to be updated.